web/client/api/geoserver/geofence/RuleService.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const axios = require('../../../libs/ajax');
+const { pickBy } = require('lodash');
+const { convertRuleGS2GF, convertRuleGF2GS } = require('../../../utils/RuleServiceUtils');
+
+
+const EMPTY_RULE = {
+    constraints: {},
+    ipaddress: "",
+    layer: "",
+    request: "",
+    rolename: "",
+    service: "",
+    username: "",
+    workspace: ""
+};
+
+/**
+ *
+ * @param {object} object the rule
+ */
+const clearNullEntries = object => pickBy(object, v => v !== null);
+
+// TODO: missing documentation
+const cleanConstraints = (rule) => {
```

Comment:
Looks like there is a lot of duplicated code, here and in the other RuleService.js

Reply:
In fact it is the differences game :smile: 

![image](https://user-images.githubusercontent.com/1279510/59520004-5ceb3580-8ec9-11e9-95a4-55cdb7f70844.png)

There are many small differences but you are right, 3 methods are equal. 

I didn't think they have to be put in common because the API on GeoServer is going to be re-implemented soon using yet another library for JavaToJson conversion. Therefore I suppose the differences will increase and so it makes sense to keep it as it is. 

What do you think?